gluten-core/src/main/scala/org/apache/gluten/softaffinity/SoftAffinityManager.scala

```diff
@@ -56,18 +55,34 @@ abstract class AffinityManager extends LogLevelUtil with Logging {
 
   lazy val detectDuplicateReading = true
 
-  lazy val maxDuplicateReadingRecords =
-    GlutenConfig.GLUTEN_SOFT_AFFINITY_MAX_DUPLICATE_READING_RECORDS_DEFAULT_VALUE
+  lazy val duplicateReadingMaxCacheItems =
+    GlutenConfig.GLUTEN_SOFT_AFFINITY_DUPLICATE_READING_MAX_CACHE_ITEMS_DEFAULT_VALUE
 
   // rdd id -> patition id, file path, start, length
-  val rddPartitionInfoMap = new ConcurrentHashMap[Int, Array[(Int, String, Long, Long)]]()
+  val rddPartitionInfoMap: LoadingCache[Integer, Array[(Int, String, Long, Long)]] =
+    CacheBuilder
+      .newBuilder()
+      .maximumSize(duplicateReadingMaxCacheItems)
```

Comment:
We also need a `expireAfterAccess` to expire old values, as the old value may not be removed if the related events are discarded.

Reply:
I thinkt this setting is not suitable here, as it assume the task need to finish in the duration we set here.

Reply:
I get. However, I still recommend adding a configuration here so that users can adjust it as needed, because for long running jobs, such as Spark Connect, old values ​​may fill up the cache, and eventually the cache will not work.

Reply:
I think the default eviction behavior is least recently used eviction, which match the expected behavior, we don't need additional settings to ensure it.
![image](https://github.com/user-attachments/assets/71ab9965-6a88-4b4c-bcf3-52b09e0244a0)

Reply:
Great, thanks for explain it.